katsdpimageutils/primarybeamcorrection/primary_beam_correction.py

```diff
@@ -1,188 +1,219 @@
     """
+    # Parse the WCS keywords in the primary HDU
     image_wcs = wcs.WCS(path)
-    # Get pointing centre of observation
+    # Get pointing centre of the observation
     phase_centre_ra = image_wcs.celestial.wcs.crval[0]
     phase_centre_dec = image_wcs.celestial.wcs.crval[1]
     # Convert to astropy.coordinates.SkyCoord object
     phase_center = coordinates.SkyCoord(phase_centre_ra, phase_centre_dec, unit=(u.deg, u.deg))
     return phase_center, image_wcs
 
 
-def radial_offset(phase_center, image_wcs, row, col):
+def radial_offset(phase_center, image_wcs, pixcrd):
+    """Determine positions (ra and dec) of the pixels and compute their
+    radial offset from phase centre in radians.
+    
+    Parameters
+    ----------
+    phase_center: astropy object
+        phase center position
+    image_wcs: atropy object
+        WCS keywords in the primary HDU
+    row: numpy array
+        An array of row pixel indices
+    pixcrd: numpy array
+        An array of column pixel coordinates
     """
-    This function gets the position in ra and dec of a pixel and determine its
-    radial offset from phase centre.
-    Input: phase center and wcs image
-    output: A separation vector between phase centre and source positions in radians.
-    """
-    p2w = image_wcs.pixel_to_world(row, col, 0, 0)[0]
+  
+    # Pixel coordinates
+    row = np.ravel(pixcrd[0])
+    col = np.ravel(pixcrd[1])
+    # Convert pixel coordinates to world coordinates
+    # The third and fourth argument is "origin" -- in this case we're declaring we
+    # have 0-based (Numpy-like) coordinates.
+    p2w = image_wcs.pixel_to_world(col, row, 0, 0)[0]
+    # A separation vector between phase centre and source positions in radians.
     separation_rad = p2w.separation(phase_center).rad
     return separation_rad
 
 
 def central_freq(path):
-    """
-    Function to get central frequency of each frequency plane.
-    Input : image header
-    Output : List of central frequency in MHz of each frequency plane.
+    """Determine central frequency of each frequency plane.
+    
+    Parameters
+    ----------
+    Input : str
+        FITS file
+    Output : numpy array
+        An array of central frequencie in MHz of each frequency plane.
     """
     images = read_fits(path)
     c_freq_plane = []
+    # NSPEC -  number of frequency planes.
     for i in range(1, images.header['NSPEC']+1):
+        # FREQ00X is the central frequency for each plane X.
         c_freq_plane.append(images.header['FREQ{0:04}'.format(i)])
     return np.array(c_freq_plane)
 
 
 def cosine_power_pattern(separation_rad, c_freq):
-    """
-    Power patterns for a given frequency based on the Cosine-squared power approximation
-    from Mauch et al. (2020).
-    Input: Radial separation array and a central frequency
+    """Compute Power patterns for a given frequency based on the
+    Cosine-squared power approximation from Mauch et al. (2020).
```

Comment:
The first line of the docstring should be a summary, and it should be followed by a blank line (PEP 257). So something like
```
Compute power patterns for a given frequency.

This uses a cosine-squared power approximation from Mauch et al. (2020).
```

Reply:
Done.